labrad/node/__init__.py

```diff
@@ -836,6 +842,49 @@ def autostart_remove(self, c, name):
             pass
         yield self.config.update_autostart(sorted(autostart))
 
+    @setting(300, returns='?')
+    def outdated_list(self, c):
+        """Get a list of servers that are outdated."""
+        outdated = self._get_outdated()
+        if outdated:
+            return tuple(tuple(info.items()) for info in outdated)
+
+    @setting(301, returns='?')
+    def outdated_restart(self, c):
+        """Restart any servers that are outdated."""
+        outdated = self._get_outdated()
+        restarts = [self.restart(c, info['instance']) for info in outdated]
+        for info, restart in zip(outdated, restarts):
+            try:
+                yield restart
+            except Exception:
+                logging.error('Failed to restart "%s"', info['instance'],
+                              exc_info=True)
+                info['status'] = 'failed to restart'
+            else:
+                info['status'] = 'restarted'
+        if outdated:
+            returnValue(tuple(tuple(info.items()) for info in outdated))
```

Comment:
This looks tough to unit test, so this might not be worth the effort (right now) -- is there any chance you can rejigger it so that some of the internal state can be injected and these methods unit tested?